lib/irc/ConnectionInstance.js

```diff
@@ -255,20 +263,43 @@ ConnectionInstance.create = function(server, opts, onCreatedCallback) {
     };
 
     var connAttempts = 0;
+    var retryTimeMs = 0;
+    var BASE_RETRY_TIME_MS = 1000;
     function retryForever() {
-        retryConnection().then(returnClient, function(err) {
+        let retryFailed = function(err) {
             connAttempts += 1;
-            var retryTimeMs = 5000; // base retry time
             if (err.message === "throttled") {
                 retryTimeMs += THROTTLE_WAIT_MS;
             }
-            // additional second for each attempt, jittered.
-            retryTimeMs += ((1000 * connAttempts) * Math.random());
 
-            setTimeout(function() {
-                retryForever();
-            }, retryTimeMs);
-        });
+            if (server.getReconnectIntervalMs() > 0) {
+                // wait until scheduled
+                Scheduler.reschedule(server, retryTimeMs, retryConnection, opts.nick).then(
+                    returnClient,
+                    retryFailed
+                );
+            }
+            else {
+                let delay = BASE_RETRY_TIME_MS + retryTimeMs +
+                           Math.round((connAttempts * 1000) * Math.random());
+
+                log.info(`Scheduling disabled, retrying with delay ${delay}ms`);
```

Comment:
This is going to be hella spammy if scheduling is disabled. I wouldn't mind if we actually logged something more useful aside from the amount of time delay (e.g the nick and server being connected, the number of `connAttempts`)